src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -584,5 +589,1762 @@ void M()
+            [|M1|](1);
+        }
+    }
+";
+            var fix_DeclarationOnly =
+@"
+    interface I1
+    {
+        void M1();
+    }
+    class C: I1
+    {
+        public void M1(int v) { }
+        void MTest() 
+        {
+            M1(1);
+        }
+    }
+";
+            var fix_All =
+@"
+    interface I1
+    {
+        void M1(int v);
+    }
+    class C: I1
+    {
+        public void M1(int v) { }
+        void MTest() 
+        {
+            M1(1);
+        }
+    }
+";
+            await TestInRegularAndScriptAsync(code, fix_DeclarationOnly, index: 0);
+            await TestInRegularAndScriptAsync(code, fix_All, index: 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_Cascading_CrossLanguage()
+        {
+            var code =
+@"
+<Workspace>
+    <Project Language=""Visual Basic"" CommonReferences=""true"" AssemblyName=""VB1"">
+        <Document FilePath=""ReferencedDocument"">
+Namespace N
+    Public Class BaseClass
+        Public Overridable Sub M()
+        End Sub
+    End Class
+End Namespace
+        </Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true""  AssemblyName=""A2"">
+        <ProjectReference>VB1</ProjectReference>
+        <Document>
+namespace N
+{
+    public class Derived: BaseClass
+    {
+        public override void M() { }
+    }
+    public class T
+    {
+        public void Test() { 
+            new Derived().[|M|](1);
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            var fix =
+@"
+<Workspace>
+    <Project Language=""Visual Basic"" CommonReferences=""true"" AssemblyName=""VB1"">
+        <Document FilePath=""ReferencedDocument"">
+Namespace N
+    Public Class BaseClass
+        Public Overridable Sub M(v As Integer)
+        End Sub
+    End Class
+End Namespace
+        </Document>
+    </Project>
+    <Project Language=""C#"" CommonReferences=""true""  AssemblyName=""A2"">
+        <ProjectReference>VB1</ProjectReference>
+        <Document>
+namespace N
+{
+    public class Derived: BaseClass
+    {
+        public override void M(int v) { }
+    }
+    public class T
+    {
+        public void Test() { 
+            new Derived().M(1);
+        }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            await TestInRegularAndScriptAsync(code, fix, index: 1);
+        }
+
+        [WorkItem(21446, "https://github.com/dotnet/roslyn/issues/21446")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TestInvocation_InvocationStyles_Positional_MoreThanOneArgumentToMuch()
+        {
+            var code =
+@"
+class C
+{
+    void M() { }
+    void Test()
+    {
+        [|M|](1, 2, 3, 4);
+    }
+}";
+            var fix0 =
+@"
+class C
+{
+    void M(int v) { }
+    void Test()
```

Comment:
why is it just add one?

Reply:
The code fix only supported adding parameters to constructors (made by @CyrusNajmabadi) when I started my work (@CyrusNajmabadi was kind enough to even document the point from where to start the work to also support normal methods). The behavior to add just one parameter at a time was already there and I hardly changed anything in the part that actually adds the new parameter to the method. We never discussed changing that behavior.